AZURE/create.yml

```diff
@@ -0,0 +1,271 @@
+        subnet_name: "{{ item.az_subnet_name }}"
+        location: "{{ item.az_location }}"
+        name: "{{ unique_vm_name[item.name] }}-vm"
+        vm_size: "{{ item.vm_size }}"
+        public_ip_allocation_method: Dynamic
+        managed_disk_type: Standard_LRS # Standard HDD, cheap :)
+        os_disk_name: "{{ unique_vm_name[item.name] }}-dsk"
+        os_type: "{{ ('windows' in item.groups) | ternary('Windows', 'Linux') }}"
+        admin_username: "{{ vm_admin_username }}"
+        admin_password: "{{ ('windows' in item.groups) | ternary(vm_admin_password, omit) }}"
+        ssh_password_enabled: "{{ ('windows' in item.groups) | ternary(omit, false) }}"
+        network_interface_names: "{{ platform_nic[item.name] }}"
+        ssh_public_keys: "{{ ('windows' in item.groups) | ternary(omit, ssh_public_keys) }}" # we have to pass param for Linux VMs only
+        image:
+          offer: "{{ (('windows' in item.groups) | ternary(azure_images.windows, azure_images.linux))[item.platform]['offer'] }}"
+          publisher: "{{ (('windows' in item.groups) | ternary(azure_images.windows, azure_images.linux))[item.platform]['publisher'] }}"
+          sku: "{{ (('windows' in item.groups) | ternary(azure_images.windows, azure_images.linux))[item.platform]['sku'] }}"
+          version: "{{ (('windows' in item.groups) | ternary(azure_images.windows, azure_images.linux))[item.platform]['version'] }}"
+        tags:
+          Instance: "{{ item.name }}"
+          CI_job: "{{ item.name }}-{{ ci_branch_name }}"
+          Name: "{{ unique_vm_name[item.name] }}-vm"
+          Created_time: "{{ lookup('pipe','date +%Y-%m-%d') }}"
+          Managed_by: "Molecule"
+          OS: "{{ (('windows' in item.groups) | ternary('Windows', 'Linux')) }}"
+          admin_user: "{{ vm_admin_username }}"
+      register: servers
+      loop: "{{ molecule_yml.platforms }}"
+      async: 7200
+      poll: 0
+
+    - name: Wait for VMs creation to complete
+      async_status:
+        jid: "{{ item.ansible_job_id }}"
+      loop: "{{ servers.results }}"
+      register: instance_create
+      until: instance_create.finished
+      delay: 20
+      retries: 300
+
+    # For Windows VMs only
+    - name: create Azure VM extension to enable HTTPS WinRM listener
+      azure_rm_virtualmachineextension:
+        name: winrm-extension
+        location: "{{ item.item.item.az_location }}"
+        resource_group: "{{ item.item.item.az_resource_group_name }}"
+        virtual_machine_name: "{{ item.ansible_facts.azure_vm.name  }}"
+        publisher: Microsoft.Compute
+        virtual_machine_extension_type: CustomScriptExtension
+        type_handler_version: 1.9
+        settings: >-
+          {'commandToExecute': 'powershell -ExecutionPolicy Unrestricted -File {{ winrm_script_name }} -EnableCredSSP',
+            'fileUris': ['{{ winrm_script }}']}
+        auto_upgrade_minor_version: true
+      loop: "{{ instance_create.results }}"
```

Comment:
Need to add async execution according to elapsed time.

Reply:
Done.